src/platform/premierp550.rs

```diff
@@ -0,0 +1,234 @@
+//! QEMU Virt board
+
+use core::fmt::Write;
+use core::{fmt, hint};
+
+use log::Level;
+use spin::Mutex;
+
+use crate::arch::{Arch, Architecture};
+use crate::config::{TARGET_FIRMWARE_ADDRESS, TARGET_START_ADDRESS};
+use crate::device::clint::{VirtClint, CLINT_SIZE};
+use crate::device::tester::{VirtTestDevice, TEST_DEVICE_SIZE};
+use crate::device::VirtDevice;
+use crate::driver::clint::ClintDriver;
+use crate::driver::uart::UartDriver;
+use crate::Platform;
+
+// —————————————————————————— Platform Parameters ——————————————————————————— //
+
+// TODO: Check about that if it is the case or not
+const MIRALIS_START_ADDR: usize = TARGET_START_ADDRESS;
+const FIRMWARE_START_ADDR: usize = TARGET_FIRMWARE_ADDRESS;
+
+// TODO: What is the base of the clint device?
+const CLINT_BASE: usize = 0x2000000;
+const TEST_DEVICE_BASE: usize = 0x3000000;
+
+// ———————————————————————————— Platform Devices ———————————————————————————— //
+
+/// The physical CLINT driver.
+///
+/// SAFETY: this is the only CLINT device driver that we create, and the platform code does not
+/// otherwise access the CLINT.
+static CLINT_MUTEX: Mutex<ClintDriver> = unsafe { Mutex::new(ClintDriver::new(CLINT_BASE)) };
+
+/// The virtual CLINT device.
+static VIRT_CLINT: VirtClint = VirtClint::new(&CLINT_MUTEX);
+
+/// The virtual test device.
+static VIRT_TEST_DEVICE: VirtTestDevice = VirtTestDevice::new();
+
+pub static WRITER: Mutex<UartDriver> = Mutex::new(UartDriver::new(
+    EIC770X_UART0_ADDR,
+    (1 << EIC770X_UART_REG_SHIFT) as usize,
+));
+
+/// The list of virtual devices exposed on the platform.
+static VIRT_DEVICES: &[VirtDevice; 2] = &[
+    VirtDevice {
+        start_addr: CLINT_BASE,
+        size: CLINT_SIZE,
+        name: "CLINT",
+        device_interface: &VIRT_CLINT,
+    },
+    VirtDevice {
+        start_addr: TEST_DEVICE_BASE,
+        size: TEST_DEVICE_SIZE,
+        name: "TEST",
+        device_interface: &VIRT_TEST_DEVICE,
+    },
```

Comment:
We can remove the test device on the board, it consumes one PMP and we don't have that many.